datapipelines/src/main/kotlin/io/customer/sdk/CustomerIO.kt

```diff
@@ -115,6 +115,15 @@ class CustomerIO private constructor(
 
         // subscribe to journey events emitted from push/in-app module to send them via data pipelines
         subscribeToJourneyEvents()
+        // if profile is already identified, republish identifier for late-added modules.
+        postProfileAlreadyIdentified()
+    }
+
+    private fun postProfileAlreadyIdentified() {
+        // TODO: Need to check SitePreferences for userId and then publish event
```

Comment:
Curious why we need to check SitePreferences? Isn't `analytics.userId()` doing the same? 🤔

Reply:
`analytics.userId()` will be set only when you call `identify` but we can have a case when you have an identifier stored that will only be in `SitePreferences` so we need to adjust the migration so that it always gets called before this method, until then we need to probably rely on SitePreferences